pkg/stanza/fileconsumer/finder.go

```diff
@@ -5,17 +5,23 @@ package fileconsumer // import "github.com/open-telemetry/opentelemetry-collecto
 
 import (
 	"github.com/bmatcuk/doublestar/v4"
+	"go.uber.org/multierr"
 )
 
 type Finder struct {
-	Include []string `mapstructure:"include,omitempty"`
-	Exclude []string `mapstructure:"exclude,omitempty"`
+	Include          []string         `mapstructure:"include,omitempty"`
+	Exclude          []string         `mapstructure:"exclude,omitempty"`
+	OrderingCriteria OrderingCriteria `mapstructure:"ordering_criteria,omitempty"`
+}
+
+type OrderingCriteria struct {
```

Comment:
@Mrod1598 and I spoke offline about this struct and whether it is necessary. I believe it is in order to allow for likely future features. Specifically:
- `group_by` would allow the user to indicate a capture field in the regex, where all the files with the same value are considered a group. This group would then be sorted individually according to the `sort_by` rules. Without a `group_by` field (as in this PR), all file names are considered one group.
- `top_n` would take the result of sorting and drop all but the top n values. Without a `top_n` value (as in this PR), we automatically select only the top 1.

@Mrod1598, if this is your understanding as well, would you create for each of these and link them in comments here?

Reply:
#23787 and #23788